src/error.rs

```diff
@@ -1,25 +1,120 @@
-use crate::git::status::Statuses;
+use crate::OidOf;
 use colored::*;
 use std::fmt;
-use std::fmt::Formatter;
-use thiserror::Error;
+use std::fmt::{Debug, Display, Formatter};
 
-#[derive(Error, Debug)]
-pub(crate) enum ErrorKind {
-    #[error("{level} - {commit_message} - ({shorthand})\n\t{cause}\n\t{additional_info}")]
+#[derive(Debug)]
+pub(crate) enum CocogittoError {
     CommitFormat {
+        oid: String,
+        summary: String,
+        author: String,
+        cause: String,
+    },
+    CommitTypeNotAllowed {
+        oid: String,
+        summary: String,
+        commit_type: String,
+        author: String,
+    },
+    NothingToCommitWithBranch {
+        branch: String,
+    },
+    NothingToCommit,
+    Semver {
+        level: String,
+        cause: String,
+    },
+    Git {
         level: String,
-        shorthand: String,
-        commit_message: String,
         cause: String,
-        additional_info: String,
     },
-    #[error("{statuses}")]
-    NothingToCommit { statuses: Statuses },
-    #[error("{level}:\n\t{cause}\n")]
-    Semver { level: String, cause: String },
-    #[error("{level}:\n\t{cause}\n")]
-    Git { level: String, cause: String },
+}
+
+impl Display for CocogittoError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            CocogittoError::CommitFormat {
+                summary,
+                oid,
+                author,
+                cause,
+            } => {
+                let error_header = "Errored commit : ".bold().red();
```

Comment:
nit: english has no space before the colon

Reply:
thanks @sfirmery, I did this mistake erywhere (resolved in adcb9ae74f71c67cc07ba5d532c6264d3b114d04)